port/win/port_win.h

```diff
@@ -81,8 +81,8 @@ namespace rocksdb {
 
 namespace port {
 
-// VS 15
-#if (defined _MSC_VER) && (_MSC_VER >= 1900)
+// VS 15 / MinGW
+#if !defined(_MSC_VER) || (_MSC_VER >= 1900)
```

Comment:
ok, I misread the code. I thought without a mingw-specific macro the code will compile for non-windows.

btw. does MSVC has `std::snprintf` in `<cstdio>`? If so we can probably update all occurrence of `snprintf` to `std::snprintf` and get rid of the macro?